src/shapes/line.class.js

```diff
@@ -285,15 +285,15 @@
    * @memberOf fabric.Line
    * @param {Object} object Object to create an instance from
    * @param {function} [callback] invoked with new instance as first argument
+   * @returns {Promise<fabric.Line>}
    */
-  fabric.Line.fromObject = function(object, callback) {
-    function _callback(instance) {
-      delete instance.points;
-      callback && callback(instance);
-    };
+  fabric.Line.fromObject = function(object) {
     var options = clone(object, true);
     options.points = [object.x1, object.y1, object.x2, object.y2];
```

Comment:
not sure why we export with x1,x2... and we import with a points array, but we need to standardize this.

Reply:
For that matter you could probably make a good argument for removing circle or even triangle as well since we have ellipse and polygon.

Reply:
what's the decision regarding the export?

Reply:
I'm not sure where did i write it, but in theory, apart circle, everything else can be done at the same speed with fabric.Path, you can still have constructors for Triangle, but will create a Path.
Then you need to write setters/getters to change the path when you change width/height. Same for line.

This PR won't take care of those things, but we can have a session where we look at those smaller classes and we decided if we want to normalize them.